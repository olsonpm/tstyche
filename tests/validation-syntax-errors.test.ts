import { afterEach, expect, test } from "@jest/globals";
import { clearFixture, writeFixture } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const isStringTestText = `import { expect, test } from "tstyche";

declare function one(a: string): void;

test("is syntax error?", () => {
  one(());
});

test("is syntax error?", () => {
  one(
});

test("is skipped?", () => {
  expect(one("abc")).type.toBeVoid();
});

test("is broken?"
`;

const tsconfig = {
  extends: "../tsconfig.json",
  include: ["./"],
};

const fixture = "validation-syntax-errors";

afterEach(async () => {
  await clearFixture(fixture);
});

test("when syntax errors are encountered", async () => {
  await writeFixture(fixture, {
    ["__typetests__/dummy.test.ts"]: isStringTestText,
    ["tsconfig.json"]: JSON.stringify(tsconfig, null, 2),
  });

  const { exitCode, stderr, stdout } = await spawnTyche(fixture);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
