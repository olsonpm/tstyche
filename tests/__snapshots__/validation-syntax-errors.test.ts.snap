// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when syntax errors are encountered: stderr 1`] = `
"Error: Expression expected. ts(1109)

  4 | 
  5 | test("is syntax error?", () => {
> 6 |   one(());
    |        ^
  7 | });
  8 | 
  9 | test("is syntax error?", () => {

      at ./__typetests__/dummy.test.ts:6:8

Error: Argument expression expected. ts(1135)

   9 | test("is syntax error?", () => {
  10 |   one(
> 11 | });
     | ^
  12 | 
  13 | test("is skipped?", () => {
  14 |   expect(one("abc")).type.toBeVoid();

       at ./__typetests__/dummy.test.ts:11:1

Error: ')' expected. ts(1005)

  16 | 
  17 | test("is broken?"
> 18 | 
     | ^

       at ./__typetests__/dummy.test.ts:18:1

"
`;

exports[`when syntax errors are encountered: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/dummy.test.ts

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
